### DISCUSSION TITLE
Next-translate issue with pre-rendered pages at build time

### DISCUSSION THREAD

**Discussion by Leonardll** [vercel/next.js#38449] [Help] [answered]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

npm run dev

### What browser are you using? (if relevant)

chrome

### How are you deploying your application? (if relevant)

vercel

### Describe the Bug

Hi there, I am having issues when I try to deploy my app using next translate. Error message is related to an array of data fetched from locales files. It work perfectly in on localhost but when I am trying to deploy it. Getting map is not function.

Does anyone have clue of how it can be fixed?
Should i use getstaticprop on the page to fix this?

import Image from "next/image";
import { AiOutlinePlus, AiOutlineMinus } from "react-icons/ai";
import { useState, useEffect } from "react";
import useTranslation from "next-translate/useTranslation";
import { Contact } from "../../components/contact/Contact";

const Conciergerie = () => {
  let { t, i18n, ready } = useTranslation("conciergerie", {
    useSuspense: true,
  });

  const scrollToConciergerie = () => {
    window.scrollTo({
      top: 300,
      behavior: "smooth",
    });
  };

  const myLoader = ({ src, width, quality }) => {
    return `${src}?w=${width}&q=${quality || 75}`;
  };

  const [showform, setshowform] = useState(false);

  useEffect(() => {
    window.addEventListener("load", scrollToConciergerie);
    return () => {
      window.removeEventListener("load", scrollToConciergerie);
    };
  });

  const showContactForm = () => {
    return <Contact />;
  };

  const contentData = t(
    "conciergerieData",
    { count: 1 },
    {
      returnObjects: true,
    }
  );

  return (
    <div className="section" onLoad={scrollToConciergerie}>
      <div className="container">
        <div className="text-center">
          <h1 className=" my-4 text-capitalize" id="conciergerie">
            {t("conciergerieHeader")}
          </h1>
        </div>
        <h3 className="text-capitalize concierge-subheading mt-3">
          {t("conciergerieTitle")}
        </h3>
        <p className="lead concierge-subheading-text">
          {t("conciergerieText")}
        </p>
      </div>
      <div className="container">
        <div className="row text-center mt-5">
          {contentData?.map((item) => {
            return (
              <div className="col-md-4" key={item.id}>
                <span className="fa-stack fa-4x">
                  <Image
                    layout="fill"
                    src={item.icon}
                    alt=""
                    className="svg-inline--fa fa-solid  fa-stack-1x fa-inverse"
                    aria-hidden="true"
                    focusable="false"
                    data-prefix="fas"
                    data-icon="house"
                    role="img"
                    loader={myLoader}
                  />
                </span>
                <h4 className="my-3 text-hogar2 text-uppercase">
                  {item.title}
                </h4>
                <ul>
                  {item.text.map((text) => {
                    return (
                      <li key={text.id} className="list-unstyled">
                        <p className="m-0 text-muted text-list">
                          {text.content}
                        </p>
                      </li>
                    );
                  })}
                </ul>

                {item.id === "algomas" &&
                  (!showform ? (
                    <AiOutlinePlus
                      role="button"
                      onClick={() => {
                        setshowform(!showform);
                      }}
                      className="fs-2"
                      fill="#5ab4ab"
                    />
                  ) : (
                    <AiOutlineMinus
                      role="button"
                      onClick={() => {
                        setshowform(!showform);
                      }}
                      className="fs-2"
                      fill="#5ab4ab"
                    />
                  ))}
                {item.id === "else" &&
                  (!showform ? (
                    <AiOutlinePlus
                      role="button"
                      onClick={() => {
                        setshowform(!showform);
                      }}
                      className="fs-2"
                      fill="#5ab4ab"
                    />
                  ) : (
                    <AiOutlineMinus
                      role="button"
                      onClick={() => {
                        setshowform(!showform);
                      }}
                      className="fs-2"
                      fill="#5ab4ab"
                    />
                  ))}
              </div>
            );
          })}
        </div>
        {showform && showContactForm()}
      </div>
    </div>
  );
};

export default Conciergerie;

please issues raise on next-translate repo 
https://github.com/vinissimus/next-translate/issues/876

### Expected Behavior

console.log({lang,locale}) should be as below 
<img width="950" alt="image" src="https://user-images.githubusercontent.com/33690476/177752360-2992f704-7ad3-4d99-8d72-d66c98b40fe6.png">

### Link to reproduction

https://github.com/Leonardll/unicoHogar/

### To Reproduce

deploy repo and see error at build time.

**Comment by Leonardll** [comment]
hi there, is there anyone that can help, please?

**Comment by Babbili** [comment, marked as answer]
@Leonardll you don't need to use any external packages, `next-translate`, to get Multilingual feature in Next Js,
Next Js has built-in support for [internationalized (i18n) routing](https://nextjs.org/docs/advanced-features/i18n-routing) via `next/router`

You can see in this [example](https://github.com/Babbili/nextjs-multilingual) i made, how to do it, it's much simpler

**Reply by Leonardll** [reply]
Hi @Babbili thanks coming to my rescue. I tried the internal library but I was having trouble mapping through a nested array. Is there any exemple for such case, please?

**Reply by Babbili** [reply]
yes after you `console.log(typeof contentData)` what did you get ?

**Reply by Leonardll** [reply]
thanks, it works but now the translation do not work in production

**Reply by Leonardll** [reply]
do you know what could cause this issue? It changes the language fine on localhost with the correct route but not in production.

**Reply by Babbili** [reply]
make sure to add i18n configuration directly to `next.config.js` file
```
const nextConfig = {

  // add the i18n config
  // this is a Sub-path Routing config, it puts the locale in the url path
  i18n: {
    locales: ['en', 'es'], // These are all the locales you want to support in your app.
    defaultLocale: 'en', // This is the default locale you want to be used when visiting a non-locale prefixed path e.g. `/contact`

    // localeDetection: false, // will no longer automatically detect which locale the user prefers based on the Accept-Language header
  },
}

module.exports = nextConfig
```
and to maintain `t` or the prefered locale across the app use `useContext` hook, checkout [appContext](https://github.com/Babbili/nextjs-multilingual/blob/main/context/AppContext.tsx)

and wrap the app [Layout](https://github.com/Babbili/nextjs-multilingual/blob/main/components/Layout.tsx) with the `contextProvider` then add the layout to [_app.js](https://github.com/Babbili/nextjs-multilingual/blob/main/pages/_app.tsx)

**Reply by Leonardll** [reply]
thanks for this @Babbili . Alright, i have tried this but it did not work.  all works perfect in development but once in production the traduction is not happening. I am gonna investigate this further. Thanks so much for you help.
